File: TempleRunner-IOS/TempleRunner-IOS/Models/ScoreModel.swift
//
//  FirstModel.swift
//  TempleRunner-IOS
//
//  Created by m2sar on 04/01/2021.
//  Copyright © 2021 UPMC. All rights reserved.
//

import UIKit

class ScoreModel {

    private var lastScore = 0
    private var hightScore  = 0
    private var currentScore = 0
    private var currentCoins = 0
    

    let urlFetch = "http://templerunnerppm.pythonanywhere.com/chat/storeScore/"
    struct Response: Codable {
        let value: String
    }

    init(){
        // Do nothing
    }

    init(lastScore: Int, hightScore: Int) {
        self.lastScore = lastScore
        self.hightScore = hightScore
    }

    init(currentScore: Int, currentCoins: Int) {
        self.currentScore = currentScore
        self.currentCoins = currentCoins
    }



    func storeHighScore() {
        print(urlFetch+Identifier.getId()+"/"+String(self.hightScore))
        let task = URLSession.shared.dataTask(with: URL( string:urlFetch+Identifier.getId()+"/"+String(self.hightScore) )!, completionHandler: {data, response, error in 
        
            if let error = error {
                print("Error accessing url: \(error)")
                return
            }     

            guard let httpResponse = response as? HTTPURLResponse,(200...299).contains(httpResponse.statusCode) else {
                print("Error with the response, unexpected status code: \(response)")
                return
            }  


            var result: Response?

            do {
                result = try JSONDecoder().decode(Response.self, from: data!)
            }
            catch {
                print("enable to aprse data")
            }

            guard let json = result else {
                print("data is nil")
                return 
            }
        
        })

        task.resume()
    }

    func getLastScore() -> Int {
        return lastScore
    }

    func getHightScore() -> Int {
        return hightScore
    }

    func getCurrentScore() -> Int {
        return currentScore
    }

    func getCurrentCoins() -> Int {
        return currentCoins
    }

    func setLastScore(val : Int)  {
        self.lastScore = val
    }

    func setHightScore(val : Int)  {
        self.hightScore = val
    }

    func setCurrentScore(val : Int)  {
        self.currentScore = val
    }

    func setCurrentCoins(val : Int)  {
        self.currentCoins = val
    }

    func updateScores() {
        if(currentScore != 0){
            lastScore = currentScore
        }
        if lastScore > hightScore {
            hightScore = lastScore
        } 
    }
    
    

}

File: TempleRunnerAndroid/app/src/main/java/com/sar/templerunner_android/Util/ScoreModel.java
package com.sar.templerunner_android.Util;

public class ScoreModel {

    private int lastScore = 0;
    private int hightScore  = 0;
    private int currentScore = 0;
    private int currentCoins = 0;


    public ScoreModel(int lastScore, int hightScore, int currentScore, int currentCoins) {
        this.lastScore = lastScore;
        this.hightScore = hightScore;
        this.currentScore = currentScore;
        this.currentCoins = currentCoins;
    }


    public ScoreModel() {
        // Do nothing
    }

    public int getLastScore() {
        return lastScore;
    }

    public void setLastScore(int lastScore) {
        this.lastScore = lastScore;
    }

    public int getHightScore() {
        return hightScore;
    }

    public void setHightScore(int hightScore) {
        this.hightScore = hightScore;
    }

    public int getCurrentScore() {
        return currentScore;
    }

    public void setCurrentScore(int currentScore) {
        this.currentScore = currentScore;
    }

    public int getCurrentCoins() {
        return currentCoins;
    }

    public void setCurrentCoins(int currentCoins) {
        this.currentCoins = currentCoins;
    }

    public void updateScores() {
        if(currentScore != 0){
            lastScore = currentScore;
        }
        if (lastScore > hightScore) {
            hightScore = lastScore;
        }
    }
}